### DISCUSSION TITLE
version upgrade, what replaced TokenMgrError?

### DISCUSSION THREAD

**Discussion by bdavis-tc** [JSQLParser/JSqlParser#1490] [Q&A] [answered]
In version 1.0 my code used this datatype in a try catch
net.sf.jsqlparser.parser.TokenMgrError

Now that I've upgraded to version 4.2 I'm curious **if there is a new class that replaced this**.  For now I can catch it as a generic Error.

I think it may have been removed between 1.0 and 1.2

**Comment by wumpz** [comment, marked as answer]
This class is generated by JavaCC using the old template. Using the modern template of JavaCC the class generated is `TokenMgrException`. I think you could be right that between 1.0 and 1.2 we switched to the modern Java generation template.
